tests/nested/core20/gadget-reseal/task.yaml

```diff
@@ -0,0 +1,51 @@
+summary: Check that a gadget refresh reseals
+
+systems: [ubuntu-20.04-64]
+
+execute: |
+    # shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB/nested.sh"
+    SEALED_KEY_MTIME_1="$(nested_exec sudo stat --format="%Y" /run/mnt/ubuntu-seed/device/fde/ubuntu-data.sealed-key)"
+    RESEAL_COUNT_1="$(nested_exec sudo cat /var/lib/snapd/device/fde/boot-chains | python3 -m json.tool | grep reseal-count|cut -f2 -d: | tr ',' ' ')"
+    # Install new (unasserted) gadget without changes and wait for change without reboot
+    boot_id="$( nested_get_boot_id )"
+    REMOTE_CHG_ID=$(nested_exec sudo snap install --dangerous /var/lib/snapd/snaps/pc_*.snap --no-wait)
+    # no reboot here, no gadget changes
+    nested_exec sudo snap watch "${REMOTE_CHG_ID}"
+  
+    # nothing in the gadget has changed so no reseal was needed
+    SEALED_KEY_MTIME_2="$(nested_exec sudo stat --format="%Y" /run/mnt/ubuntu-seed/device/fde/ubuntu-data.sealed-key)"
+    test "$SEALED_KEY_MTIME_2" -eq "$SEALED_KEY_MTIME_1"
+    RESEAL_COUNT_2="$(nested_exec sudo cat /var/lib/snapd/device/fde/boot-chains | python3 -m json.tool | grep reseal-count|cut -f2 -d: | tr ',' ' ')"
+    test "$RESEAL_COUNT_2" = "$RESEAL_COUNT_1"
+
+    # create modified boot assets
+    KEY_NAME=$(nested_get_snakeoil_key)
+    SNAKEOIL_KEY="$PWD/$KEY_NAME.key"
+    SNAKEOIL_CERT="$PWD/$KEY_NAME.pem"
+    # ensure clean pc-gadget dir
+    rm -rf pc-gadget
+    snap download --basename=pc --channel="20/edge" pc
+    unsquashfs -d pc-gadget pc.snap
+    # change a few bytes in the compat header
+    sed -i 's/This program cannot be run in DOS mode/This program cannot be run in XXX mode/' pc-gadget/grubx64.efi
```

Comment:
I think we should have a guarding `grep -a -q This program cannot be run in DOS mode` before to make sure that if the string ever changes such that our replacement fails, we don't keep running the rest of the test since otherwise sed will just silently go on if the replacement doesn't happen

Reply:
fine for a followup btw